### PR TITLE
[FIX] web: kanban: quick create in a dialog

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: addons/web/static/src/views/kanban/kanban_renderer.js [hunk 1]
```diff
@@ -377,8 +377,16 @@ export class KanbanRenderer extends Component {
                         resModel: this.props.list.resModel,
                         context,
                         title: this.env._t("Create"),
-                        onRecordSaved: (record) => {
-                            group.addRecord(record, 0);
+                        onRecordSaved: async (record) => {
+                            await group.list.createRecord(
```
Comment (Polymorphe57) [inline addons/web/static/src/views/kanban/kanban_renderer.js:381]: group.createRecord for count+++

File: addons/web/static/src/views/kanban/kanban_renderer.js [hunk 2]
```diff
@@ -388,11 +389,20 @@ export class KanbanRenderer extends Component {
                         context,
                         title: this.env._t("Create"),
                         onRecordSaved: (record) => {
-                            group.addRecord(record, 0);
+                            prom = group.createRecord(
```
Comment (aab-odoo) [inline addons/web/static/src/views/kanban/kanban_renderer.js:392]: isn't it enough to await here? It seems that the FormViewDialog waits for the promise before closing the dialog

File: addons/web/static/src/views/relational_model.js [hunk 3]
```diff
@@ -1796,9 +1799,10 @@ export class DynamicRecordList extends DynamicList {
             await this.load();
         }
         await this.model.keepLast.add(this.model.mutex.exec(() => newRecord.load()));
-        this.editedRecord = newRecord;
-        this.onRemoveNewRecord = await this.onCreateRecord(newRecord);
-
+        if (mode === "edit") {
+            this.editedRecord = newRecord;
+            this.onRemoveNewRecord = await this.onCreateRecord(newRecord);
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model.js:1804]: not sure about this not being called if mode is readonly

**Review by aab-odoo — APPROVED** [review]
I took some time to read this PR, and now that I better understand it, I think it's not that bad :D Yet, I believe that the original purpose of  `createRecord` wasn't taking this usecase into account (which doesn't mean we can't make it deal with it now). Also, there's another mechanism for the "basic" quick create (only display name) to be correctly reloaded with the kanban fields: we set `parentActiveFields`. We should probably remove this someday, and make that flow use the path you introduce here.

File: addons/web/static/src/views/kanban/kanban_renderer.js (hunk 2 above)
Comment (Polymorphe57) [reply addons/web/static/src/views/kanban/kanban_renderer.js:392]: ok right

File: addons/web/static/src/views/relational_model.js (hunk 3 above)
Comment (Polymorphe57) [reply addons/web/static/src/views/relational_model.js:1804]: add a method addExistingRecord

**Review by aab-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Pull request closed without merging**